File: service-auth-node/auth/jwt.js
const jwt = require('jsonwebtoken');

const SECRET = process.env.JWT_SECRET;

if (!SECRET) {
	throw new Error('JWT_SECRET not set');
}

function generateToken(userId) {
	return jwt.sign({ userId }, SECRET, { expiresIn: '1h' });
}

function verifyToken(token) {
	return jwt.verify(token, SECRET);
}

module.exports = { generateToken, verifyToken };

File: service-ledger-go/auth/jwt.go
package auth

import (
	"errors"
	"os"

	"github.com/golang-jwt/jwt/v5"
)

func VerifyToken(tokenString string) (jwt.MapClaims, error) {
	secret := os.Getenv("JWT_SECRET")
	if secret == "" {
		return nil, errors.New("JWT_SECRET not set")
	}

	token, err := jwt.Parse(tokenString, func(t *jwt.Token) (interface{}, error) {
		return []byte(secret), nil
	})

	if err != nil || !token.Valid {
		return nil, errors.New("invalid token")
	}

	return token.Claims.(jwt.MapClaims), nil
}